app/models/generic_object.rb

```diff
@@ -14,13 +14,11 @@ class GenericObject < ApplicationRecord
     :string   => ActiveRecord::Type::String.new,
   }.freeze
 
-  def must_be_defined_attributes
-    return errors.add(:base, "must specify a GenericObjectDefinition.") unless generic_object_definition
-
-    found = custom_attributes.detect { |ca| !generic_object_definition.defined_attributes.include?(ca.name) }
-    if found
-      errors.add(:base, "#{found.name} is not defined in GenericObjectDefinition: #{generic_object_definition.defined_attributes.keys.join(", ")}.")
-    end
+  def initialize(attributes = {})
+    attributes = (attributes || {}).stringify_keys
+    attributes = attributes.slice('generic_object_definition').merge(attributes.except('generic_object_definition'))
+    # generic_object_definition will be set first since hash iteration is based on the order of key insertion
```

Comment:
Can you put this comment *above* the code line?  It feels backwards to have the explanation afterwards.